### ISSUE TITLE
Line endings 

### ISSUE THREAD

**Issue by Sylwekqaz** [madskristensen/WebCompiler#136]
Web compiler ignore line ending from orginal file 
for example my .less files end with CRLF and after compilation .css files have LF endings.
Its anoying because git always see changes in compiled files.

**Comment by madskristensen** [comment]
I think this is beyond the scope of the extension. Not sure. Have you looked at controlling that using `.gitattributes`? https://help.github.com/articles/dealing-with-line-endings/

**Comment by Sylwekqaz** [comment]
Yes, I have set in `.gitattributes` normalization to `CLRF` in whole project. I think that extension should keep end of line style from source file.

**Comment by madskristensen** [comment]
The Sass compiler has an option for this, but the LESS compiler does not. This extension could change the line endings of LESS files from C# though. Why do you commit the output files to source control? Wouldn't it be better not to do that and thus avoiding this issue completely?

**Comment by Sylwekqaz** [comment]
I try exclude output files on my private project. I added `.css` and `.map` to git ignore, and delete them from source control. When friend compiles project extension regenerate `.css` `.min.css` files but no `.map` files. Extension add reference to `.map` files to project. Deployment site (by webdeploy) fails with error

      Copying file {filename} to {path}\{filename} failed. Could not find file {filename}.

**Comment by madskristensen** [comment]
Is this issue with the .map file a separate issue?

**Comment by Sylwekqaz** [comment]
Yes

**Comment by tjrobinson** [comment]
I am also having problems with the output file line endings being different to the source file. Using Sass in this case.

The problem is that the "Inconsistent Line Endings" dialog keeps being shown. I can disable it but I find it useful for other files.

![image](https://cloud.githubusercontent.com/assets/925311/16154273/fbd872aa-34a1-11e6-90cc-cf1f7034c76f.png)

**Comment by RobCrockett** [comment]
I am seeing the same problem with the SASS compiler.  We have not yet setup WebCompiler on our build  machine so we have the CSS files in source control.  When I touch a SASS file WebCompiler compiles it and I get an output file with LF rather CRLF.  This is not a problem with source control.  The version in Git has the correct line ending.  Only the newly created CSS file has the incorrect line endings.

I tried setting the lineFeed compiler option but that did not help.  It appears to default to CRLF anyway.

**Comment by os1r1s110** [comment]
Has there been any solution provided yet for that little issue? I am facing the exact same problem with Visual Studio 2015 and web compiler. Whenever I modify my main.scss file for example, the css file generated will have inconsistent line endings... I am trying to find a solution but to no avail yet... 

Thanks for any help!

**Comment by os1r1s110** [comment]
It seems like my problem is caused by block comments in the scss files.

`/* ================================================== 
    Example block comment
  ================================================== */
`

If I remove them from my scss, then Visual Studio doesn't complain anymore about inconsistent line endings. The thing is I still don't understand why a block comment would generate those inconsistent line endings..... 

Any idea?

**Comment by levininja** [comment]
I don't think the merged PR actually addressed this issue.  If so it's not documented well.  What I'm inferring is that I should be able to change an entry in compilerconfig.json to have this option in it, but this change has no effect on line endings:

```
		"options": {
			"linefeed":  "crlf",
		}
```

if that isn't how it's supposed to work please inform.

**Comment by AndreyLapitsky** [comment]
> I don't think the merged PR actually addressed this issue. If so it's not documented well. What I'm inferring from the PR is that I should be able to change an entry in compilerconfig.json to have this option in it, but this change has no effect on line endings:
> 
> ```
> 		"options": {
> 			"linefeed":  "crlf",
> 		}
> ```
> if that isn't how it's supposed to work please inform.

It's really fixed :) Just type "lineFeed" instead "linefeed"

**Comment by pankajparkar** [comment]
@AndreyLapitsky and @levininja you both saved my day. Thanks.
